lms/djangoapps/course_api/courseware_access.py

```diff
@@ -0,0 +1,109 @@
+""" Centralized access to LMS courseware app """
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+from courseware import courses, module_render
+from courseware.model_data import FieldDataCache
+
+
+def get_course(course_id, depth=0):
+    """
+    Utility method to obtain course components
+    """
+    course_descriptor = None
+    course_content = None
+    course_key = CourseKey.from_string(course_id)
+    if course_key:
+        course_descriptor = get_course_descriptor(course_key, depth)
+    return course_descriptor, course_key, course_content
```

Comment:
Also, course_key can be easily retrieved from the course_descriptor (via the id property), so it's unnecessary to return it as a separate tuple member